### ISSUE TITLE
Specs: `Environment %s is referenced in function %s but not declared in specs`

### ISSUE THREAD

**Issue by Altair-Bueno** [fission/fission#2683]
**Fission/Kubernetes version**

<pre>
$ fission version
client:
  fission/core:
    BuildDate: "2022-09-16T13:24:57Z"
    GitCommit: b36e0516
    Version: v1.17.0
server:
  fission/core:
    BuildDate: "2022-09-16T13:24:57Z"
    GitCommit: b36e0516
    Version: v1.17.0
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.6", GitCommit:"d921bc6d1810da51177fbd0ed61dc811c5228097", GitTreeState:"clean", BuildDate:"2021-10-27T17:50:34Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.6", GitCommit:"d921bc6d1810da51177fbd0ed61dc811c5228097", GitTreeState:"clean", BuildDate:"2021-10-27T17:44:26Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}
</pre>

**Kubernetes platform (e.g. Google Kubernetes Engine)**

**Describe the bug**

template string on warning message

**To Reproduce**

Run `fission spec validate` with the following sample code

[redis-glue-newdeploy.zip](https://github.com/fission/fission/files/10372270/redis-glue-newdeploy.zip)

**Expected result**

`%s` should be replaced for some value

**Actual result**

**Screenshots/Dump file**

```sh
$ fission spec validate
DeployUID: 818a6fb7-08c5-46bc-a156-56dd1c72cfa4
Resources:
 * 1 Functions
 * 0 Environments
 * 1 Packages
 * 1 Http Triggers
 * 0 MessageQueue Triggers
 * 0 Time Triggers
 * 0 Kube Watchers
 * 1 ArchiveUploadSpec
Warning: Environment %s is referenced in function %s but not declared in specs
Warning: python-newdeploy
Warning: redis-glue-newdeploy
Validation Successful
```

**Additional context**

**Comment by Altair-Bueno** [comment]
# Additional info:

`Environment %s is referenced in function %s but not declared in specs` can be found here

https://github.com/fission/fission/blob/496e4e31623d9febfb0ace568290bacb58d8ae72/pkg/fission-cli/cmd/spec/spec.go#L487

**Comment by NikhilSharmaWe** [comment]
@Altair-Bueno This issue is resolved in the `v1.18` version through this [commit](https://github.com/fission/fission/commit/b9513868ed9617e3c3e6e779c37878d4a0030c60).

**Comment by Altair-Bueno** [comment]
Thanks, i'll update as soon as possible

**Comment by sanketsudake** [comment]
Closing issue as it is resolved in 1.18. Feel free to reopen if you observe again.
